src/test/java/org/sonarsource/sonarlint/ls/mediumtests/LanguageServerMediumTests.java

```diff
@@ -331,12 +330,37 @@ void noAnalysisOnNullContent() throws Exception {
     lsProxy.getTextDocumentService()
       .didClose(new DidCloseTextDocumentParams(docId));
 
+    assertThat(client.getDiagnostics(uri)).isNull();
+  }
+
+  @Test
+  void vcsIgnoredShouldNotAnalyzed() throws Exception {
+    emulateConfigurationChangeOnClient("**/*Test.js", true, true, true);
+    Thread.sleep(1000);
+    client.logs.clear();
+
+    String uri = getUri("foo.py");
+    client.diagnosticsLatch = new CountDownLatch(1);
+    client.isIgnoredByScm = true;
+    VersionedTextDocumentIdentifier docId = new VersionedTextDocumentIdentifier(uri, 1);
+
+    // SLVSCODE-157 - Open/Close/Open/Close triggers a race condition that nullifies content
+    lsProxy.getTextDocumentService()
+            .didOpen(new DidOpenTextDocumentParams(new TextDocumentItem(uri, "python", 1, "# Nothing to see here\n")));
+    lsProxy.getTextDocumentService()
+            .didClose(new DidCloseTextDocumentParams(docId));
+    lsProxy.getTextDocumentService()
+            .didOpen(new DidOpenTextDocumentParams(new TextDocumentItem(uri, "python", 1, "# Nothing to see here\n")));
+    lsProxy.getTextDocumentService()
+            .didClose(new DidCloseTextDocumentParams(docId));
+
+
     await().atMost(1, TimeUnit.MINUTES).untilAsserted(
-      () -> assertThat(client.logs).extracting(withoutTimestamp()).contains(
-        "[Debug] Skipping analysis of file '" + uri + "', content has disappeared"
-      )
+            () -> assertThat(client.logs).extracting(withoutTimestamp())
+                    .contains("Skip analysis for SCM ignored file: " + uri)
     );
     assertThat(client.getDiagnostics(uri)).isEmpty();
+    client.isIgnoredByScm = false;
```

Comment:
Would be better to cleanup in the `@BeforeEach` or `@AfterEach` methods, so that the cleanup happen even if the test fails in the middle.